### ISSUE TITLE
508-defect-4 [MOBILE DESIGN, ZOOM]: When page is zoomed in, content SHOULD not be cut off

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#8414]
# [508-defect-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

As part of the accessibility review process, an auditor zooms to 200%, 300%, and 400% with the screen at 1280px, to verify content is usable and not cut off. At 200% and 400%, there are some minor cut off issues. 

* At 200%, it is more of a cosmetic issue when the alert is overlapped by the map.
* At 400%, the form and the map overlap the search results, and with the downtime banner things get very tight.

This may be more appropriate for the ux/ui improvements, but be more evident due to the alert.

### Why this matters

Some people with low-vision use screen magnifiers, which may be browser extensions, operating system level utilities, or even external tools that attach to a monitor, often in addition the browser zoom. Ensuring that content is still usable when zoomed in using any or a combination of these is crucial for these individuals. If content is obscured or cut off them may not be able to access what they need. This is generally a good way of validating a mobile experience, where people may have zoom added, or the device have a range of dimensions and resolutions.

Please note: Note: 320 CSS pixels is equivalent to a starting viewport width of 1280 CSS pixels wide at 400% zoom. For web content which are designed to scroll horizontally (e.g. with vertical text), the 256 CSS pixels is equivalent to a starting viewport height of 1024px at 400% zoom.

Content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions for:

Vertical scrolling content at a width equivalent to 320 CSS pixels;
Horizontal scrolling content at a height equivalent to 256 CSS pixels.
Except for parts of the content which require two-dimensional layout for usage or meaning.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a magnification tool user, I want to access all content and functionality without it being obscured or cut off.

## Environment

* Operating System: any
* Browser: any
* Server destination: production

## Steps to Recreate

1. Enter this url in browser (Chrome recommended for this): https://www.va.gov/find-locations/?address=Washington%2C%20DC&context=Washington%2C%20District%20of%20Columbia%2C%20United%20States&facilityType=benefits&location=38.895%2C-77.0366&page=1&serviceType&zoomLevel=4
2. Open Developer Tools, and make sure the viewport width is 1280px wide
3. Using the browser zoom, zoom to 200%, 300%, and 400%
4. Verify what is shown in the screen shots below, and experiment with scrolling a bit to explore the ux

## Possible Fixes (optional)

Ensure the CSS styling prevents overlap and cut off, particularly in the masonry.

* [W3C WCAG 2.0 F69 - Failure of Success Criterion 1.4.4 when resizing visually rendered text up to 200 percent causes the text, image or controls to be clipped, truncated or obscured](https://www.w3.org/TR/WCAG20-TECHS/F69.html)

* [W3C WCAG - Success Criterion 1.4.10 Reflow](https://www.w3.org/TR/WCAG21/)

## Screenshots or Trace Logs

![200pct](https://user-images.githubusercontent.com/57469/80250107-2893ea00-8641-11ea-8d4b-40e9133f7ab6.png)

![400pct-01](https://user-images.githubusercontent.com/57469/80250114-2b8eda80-8641-11ea-99dd-5add6965aecc.png)

[Movie example of 400% UX, located in Slack](https://dsva.slack.com/archives/C0FQSS30V/p1587756980458300?thread_ts=1587754521.457400&cid=C0FQSS30V)

**Comment by mmiddaugh** [comment]
This issue should be resolved by our recent responsive design changes. Please validate at your convenience, @jenstrickland

**Comment by jenstrickland** [comment]
I tested in production, and it isn't fixed. Should I test in staging?

![Screen Shot 2020-08-26 at 5 09 28 PM](https://user-images.githubusercontent.com/57469/91357317-0c56b400-e7bf-11ea-967c-10de37d2bd17.png)


![Screen Shot 2020-08-26 at 5 09 48 PM](https://user-images.githubusercontent.com/57469/91357323-0fea3b00-e7bf-11ea-9ecc-5c5c828f6a17.png)


![Screen Shot 2020-08-26 at 5 09 54 PM](https://user-images.githubusercontent.com/57469/91357332-137dc200-e7bf-11ea-9011-e9da84de405c.png)

**Comment by mmiddaugh** [comment]
@jenstrickland this is ready for validation when you are available. Thank you!

**Comment by jenstrickland** [comment]
1. On the bottom of the map container there is a `margin-bottom: 24px` that is causing the text to wrap improperly. Once this is addressed, this ticket can be closed.


![Screen Shot 2020-09-08 at 6 14 49 PM](https://user-images.githubusercontent.com/57469/92532873-4c765780-f1ff-11ea-9f3d-0c0c866673b7.png)

**Potential fix:**
Remove the `margin-bottom: 24px` from the `.desktop-map-container`
 
```diff
<!-- Current Code -->
.facility-locator .desktop-map-container {
    min-height: 78vh;
    padding-left: 0;
    float: right;
    width: calc(100% - 305px);
!    margin-bottom: 24px;
}
```

![Screen Shot 2020-09-08 at 6 22 03 PM](https://user-images.githubusercontent.com/57469/92533325-39b05280-f200-11ea-812f-6f371a589416.png)



<hr/>
2. In the [VA Design System](https://design.va.gov/components/form-controls), form controls have a max-width of 46rem. Although the buttons documentation does not specify a max-width, it would be best practice that buttons would also have this max-width. @mmiddaugh — would you prefer I open a new ticket for this item, or is this on the team's radar?

![Screen Shot 2020-09-08 at 6 14 17 PM](https://user-images.githubusercontent.com/57469/92532815-20f36d00-f1ff-11ea-959d-6ba949df27cb.png)

**Comment by mmiddaugh** [comment]
Jennifer- thank you for your input above. 

1. The text wrapping situation was identified by the Facilities team and resolved separately.

2. Button width was not part of the original issue. As it is not a defined requirement in the design system,  I've created a [separate issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13692) to consider your recommendation during a future iteration. 

The original issue of obscured/cut-off content is resolved.
